### ISSUE TITLE
rebuild realm-js fail for electron

### ISSUE THREAD

**Issue by zhujiaming** [realm/realm-js#2387]
## Goals

npm install --save realm
and rebuid realm for electron

## Expected Results

rebuild realm success

## Actual Results
```

D:\workspace_0\electron\electron-learn-1>.\node_modules\.bin\electron-rebuild.cmd .\node_modules\ffi\
× Rebuild Failed

An unhandled error occurred inside electron-rebuild
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.15.0 | win32 | x64
gyp info spawn C:\Users\jm\.windows-build-tools\python27\python.EXE
gyp info spawn args [ 'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2015',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\jm\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\jm\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\Users\\jm\\.electron-gyp\\4.0.2\\<(target_arch)\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\build',
gyp info spawn args   '-Goutput_dir=.' ]
Warning: Missing input files:
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\reporters.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\support\jasmine.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\asserts.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\publish.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\async-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\linkingobjects-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\ava\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\worker-tests-script.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\adapter-tests.ts
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\jest\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\admin-user-helper.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\results-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\schemas.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\jest\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\object-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\mocha\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\user-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\setup-module-path.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\unit_tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\object-id-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\worker.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\encryption-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\mock_realm.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\ava\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\list-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\changelog-header.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\notifier-tests.ts
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\session-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\download-api-helper.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\query-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\mocha\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\test.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\realm-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\index.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\index.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\.eslintrc.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\set-version.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\garbage-collection.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\permission-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\query-tests.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\migration-tests.js
gyp info spawn C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
�ڴ˽��������һ������һ����Ŀ����Ҫ���ò������ɣ������ӡ�/m�����ء�
  download-realm
  Resolved requirements: { SYNC_SERVER_FOLDER: 'sync',
    SYNC_ARCHIVE: 'realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz',
    CORE_SERVER_FOLDER: 'core/v5.19.1/windows/x64/nouwp/Release',
    CORE_ARCHIVE: 'realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz' }
  No lockfile found at the target, proceeding.
  Download url: https://static.realm.io/downloads/core/v5.19.1/windows/x64/nouwp/Release/realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz
  Downloading realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz
  Extracting realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz => D:\workspace_0\electron\electron-learn-1\node_modules\realm\vendor\realm-win-x64
  Download url: https://static.realm.io/downloads/sync/realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz
  Downloading realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz
  Extracting realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz => D:\workspace_0\electron\electron-learn-1\node_modules\realm\vendor\realm-win-x64
  Success
  binding_callback_thread_observer.cpp
  collection_notifications.cpp
  index_set.cpp
  list.cpp
  object.cpp
  object_schema.cpp
  object_store.cpp
  placeholder.cpp
  results.cpp
  schema.cpp
  shared_realm.cpp
  thread_safe_reference.cpp
  collection_change_builder.cpp
  collection_notifier.cpp
  list_notifier.cpp
  object_notifier.cpp
  primitive_list_notifier.cpp
  realm_coordinator.cpp
  results_notifier.cpp
  transact_log_handler.cpp
  weak_realm_notifier.cpp
  uuid.cpp
  external_commit_helper.cpp
  adapter.cpp
  admin_realm.cpp
  global_notifier.cpp
  sync_file.cpp
  sync_metadata.cpp
  work_queue.cpp
  partial_sync.cpp
  sync_config.cpp
  sync_manager.cpp
  sync_session.cpp
  sync_user.cpp
  win_delay_load_hook.cc
  object-store.vcxproj -> D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\Release\\object-store.lib
  js_realm.cpp
  node_init.cpp
  platform.cpp
  sync_logger.cpp
  win_delay_load_hook.cc
LINK : fatal error LNK1181: �޷��������ļ���C:\src\vcpkg\installed\x64-windows-static\lib\libeay32.lib�� [D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\realm.vcxproj]
gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\workspace_0\electron\electron-learn-1\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "D:\\softs\\nodejs\\node.exe" "D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=4.0.2" "--arch=x64" "--dist-url=https://atom.io/download/electron" "--build-from
-source" "--module_name=realm" "--module_path=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\compiled\\electron-v4.0_win32_x64" "--host=https://static.realm.io" "--remote_path=/node-pre-gyp/2.28.0/"
gyp ERR! cwd D:\workspace_0\electron\electron-learn-1\node_modules\realm
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 

Failed with exit code: 1

Error: gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.15.0 | win32 | x64
gyp info spawn C:\Users\jm\.windows-build-tools\python27\python.EXE
gyp info spawn args [ 'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2015',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\jm\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\jm\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\Users\\jm\\.electron-gyp\\4.0.2\\<(target_arch)\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\build',
gyp info spawn args   '-Goutput_dir=.' ]
Warning: Missing input files:
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\reporters.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\support\jasmine.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\asserts.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\publish.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\async-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\linkingobjects-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\ava\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\worker-tests-script.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\adapter-tests.ts
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\jest\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\admin-user-helper.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\results-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\schemas.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\jest\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\object-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\mocha\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\user-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\setup-module-path.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\unit_tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\object-id-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\worker.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\encryption-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\spec\helpers\mock_realm.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\ava\test.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\list-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\changelog-header.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\notifier-tests.ts
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\session-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\download-api-helper.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\query-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\test-runners\mocha\package.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\test.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\realm-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\index.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\index.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\.eslintrc.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\scripts\set-version.sh
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\garbage-collection.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\permission-tests.js
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\query-tests.json
D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\..\tests\js\migration-tests.js
gyp info spawn C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
�ڴ˽��������һ������һ����Ŀ����Ҫ���ò������ɣ������ӡ�/m�����ء�
  download-realm
  Resolved requirements: { SYNC_SERVER_FOLDER: 'sync',
    SYNC_ARCHIVE: 'realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz',
    CORE_SERVER_FOLDER: 'core/v5.19.1/windows/x64/nouwp/Release',
    CORE_ARCHIVE: 'realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz' }
  No lockfile found at the target, proceeding.
  Download url: https://static.realm.io/downloads/core/v5.19.1/windows/x64/nouwp/Release/realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz
  Downloading realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz
  Extracting realm-core-Release-v5.19.1-Windows-x64-devel.tar.gz => D:\workspace_0\electron\electron-learn-1\node_modules\realm\vendor\realm-win-x64
  Download url: https://static.realm.io/downloads/sync/realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz
  Downloading realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz
  Extracting realm-sync-Release-v4.4.2-Windows-x64-devel.tar.gz => D:\workspace_0\electron\electron-learn-1\node_modules\realm\vendor\realm-win-x64
  Success
  binding_callback_thread_observer.cpp
  collection_notifications.cpp
  index_set.cpp
  list.cpp
  object.cpp
  object_schema.cpp
  object_store.cpp
  placeholder.cpp
  results.cpp
  schema.cpp
  shared_realm.cpp
  thread_safe_reference.cpp
  collection_change_builder.cpp
  collection_notifier.cpp
  list_notifier.cpp
  object_notifier.cpp
  primitive_list_notifier.cpp
  realm_coordinator.cpp
  results_notifier.cpp
  transact_log_handler.cpp
  weak_realm_notifier.cpp
  uuid.cpp
  external_commit_helper.cpp
  adapter.cpp
  admin_realm.cpp
  global_notifier.cpp
  sync_file.cpp
  sync_metadata.cpp
  work_queue.cpp
  partial_sync.cpp
  sync_config.cpp
  sync_manager.cpp
  sync_session.cpp
  sync_user.cpp
  win_delay_load_hook.cc
  object-store.vcxproj -> D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\Release\\object-store.lib
  js_realm.cpp
  node_init.cpp
  platform.cpp
  sync_logger.cpp
  win_delay_load_hook.cc
LINK : fatal error LNK1181: �޷��������ļ���C:\src\vcpkg\installed\x64-windows-static\lib\libeay32.lib�� [D:\workspace_0\electron\electron-learn-1\node_modules\realm\build\realm.vcxproj]
gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\workspace_0\electron\electron-learn-1\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "D:\\softs\\nodejs\\node.exe" "D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=4.0.2" "--arch=x64" "--dist-url=https://atom.io/download/electron" "--build-from
-source" "--module_name=realm" "--module_path=D:\\workspace_0\\electron\\electron-learn-1\\node_modules\\realm\\compiled\\electron-v4.0_win32_x64" "--host=https://static.realm.io" "--remote_path=/node-pre-gyp/2.28.0/"
gyp ERR! cwd D:\workspace_0\electron\electron-learn-1\node_modules\realm
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 

Failed with exit code: 1
    at SafeSubscriber._error (D:\workspace_0\electron\electron-learn-1\node_modules\spawn-rx\lib\src\index.js:267:84)
    at SafeSubscriber.__tryOrUnsub (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.error (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:156:26)
    at Subscriber._error (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:92:26)
    at Subscriber.error (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:72:18)
    at MapSubscriber.Subscriber._error (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:92:26)
    at MapSubscriber.Subscriber.error (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:72:18)
    at SafeSubscriber._next (D:\workspace_0\electron\electron-learn-1\node_modules\spawn-rx\lib\src\index.js:242:65)
    at SafeSubscriber.__tryOrUnsub (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.next (D:\workspace_0\electron\electron-learn-1\node_modules\rxjs\internal\Subscriber.js:143:22)

```

It looks like can not find the file ('C:\src\vcpkg\installed\x64-windows-static\lib\libeay32.lib')
who can help me? thanks

## Steps to Reproduce
```
 npm config set msvs_version 2015
 .\node_modules\.bin\electron-rebuild.cmd
```

## Version of Realm and Tooling

- Realm JS SDK Version: ^2.28.0
- Node:  10.15.0
- Electron Version: 4.2.0

**Comment by rajivshah3** [comment]
I think we ran into this issue as well when we built Realm for our Electron app. We fixed it by running the following:

```
git clone https://github.com/Microsoft/vcpkg C:\src\vcpkg
cd C:\src\vcpkg
.\bootstrap-vcpkg.bat
.\vcpkg install openssl:x64-windows-static
```
It installs the OpenSSL VC++ Static 64bit Library from https://github.com/Microsoft/vcpkg which seems to be a dependency of Realm on Windows

**Comment by zhujiaming** [comment]
> I think we ran into this issue as well when we built Realm for our Electron app. We fixed it by running the following:
> 
> ```
> git clone https://github.com/Microsoft/vcpkg C:\src\vcpkg
> cd C:\src\vcpkg
> .\bootstrap-vcpkg.bat
> .\vcpkg install openssl:x64-windows-static
> ```
> 
> It installs the OpenSSL VC++ Static 64bit Library from https://github.com/Microsoft/vcpkg which seems to be a dependency of Realm on Windows

thanks ,it works!

**Comment by aklinker1** [comment]
I'm also getting this error, but installing `openssh` from `vcpkg` did not resolve the issue.

Building it creates a `libcrypto.lib` (instead of `libeay32.lib`) and `libssl.lib` (instead of `ssleay32.lib`). Apparently OpenSSL changed these in version 1.1.0. 

- I tried installing a version of 1.0.2 from an archive, but that has .dll not .lib files for what realm is looking for, and I don't know how to convert between the two.
- I tried installing a earlier version from `vcpkg` by checking out a hash from May 1, 2019 (because thats how vcpkg manages versions...), but it couldn't install openssl from there
- I tried renaming the files above to what realm wants, but that didn't work either because of internal references that are wrong

Any ideas?
